lib/bounces.js

```diff
@@ -108,6 +108,24 @@ module.exports = function (log, error) {
           }
         }
 
+        // Log flow metrics if `flowId` and `flowBeginTime` specified in headers
+        const flowId = getHeaderValue('X-Flow-Id', message)
+        const flowBeginTime = getHeaderValue('X-Flow-Begin-Time', message)
+        if (flowId && flowBeginTime) {
+          const eventName = 'email.' + templateName + '.bounced'
+          const currentTime = Date.now()
+
+          // Matches structure emitted by https://github.com/mozilla/fxa-auth-server/blob/master/lib/metrics/context.js#L122
+          const flowEventInfo = {
+            event: eventName,
+            time: currentTime,
+            flow_id: flowId,
+            flow_time: (currentTime <= flowBeginTime) ? 0 : (currentTime - flowBeginTime)
+          }
+
+          log.info('flowEvent', flowEventInfo)
```

Comment:
I get why you opted for this approach, because it saves the hassle of faking the request object required to call `log.flowEvent` directly. But I wonder whether it is worth moving this call and the preceding assignment to `flowEventInfo` into a function inside `lib.log.js`?

I have a slight concern that we're doing the same thing in two places, albeit it's not a huge amount. If the current `log.flowEvent` was renamed to something along the lines of `log.flowEventFromRequest`, it could call `log.flowEvent` to do this stuff and your code could do the same.

What do you think, worth it or am I dreaming up imaginary obstacles?